didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+                "type": "DIDCommMessaging",
+                "serviceEndpoint": {
+                    "uri": relay_did,
+                    "accept": ["didcomm/v2"],
+                },
+            }
+        ],
+    )
+    LOG.info("relayed did: ", new_did)
+
+    # A couple of helpers variables to simplify the next few lines
+    resolver = dmp.resolver
+    secrets = dmp.secrets
+
+    # Add the DID to our secrets manager so that we can unpack messages
+    # destined to us via our new DID
+    doc = await resolver.resolve_and_parse(new_did)
+    # New format, key-# is in order of keys in did
+    await secrets.add_secret(AskarSecretKey(keys[0], f"{new_did}#key-1"))
+    await secrets.add_secret(AskarSecretKey(keys[1], f"{new_did}#key-2"))
+
+    # Legacy formats
+    # verkey
+    await secrets.add_secret(AskarSecretKey(keys[0], doc.authentication[0]))
+    # xkey
+    await secrets.add_secret(AskarSecretKey(keys[1], doc.key_agreement[0]))
+
+    # Send a message to the relay informing it of our new endpoint that people
+    # should contact us by
+    message = Message(
+        type="https://didcomm.org/coordinate-mediation/3.0/recipient-update",
+        id=str(uuid.uuid4()),
+        body={
+            "updates": [
+                {
+                    "recipient_did": new_did,
+                    "action": "add",
+                },
+            ],
+        },
+        frm=my_did,
+        to=[relay_did],
+    )
+    message = await send_http_message(dmp, my_did, message, target=relay_did)
+
+    return new_did
+
+
+async def fetch_relayed_messages(
+    dmp: DIDCommMessaging,
+    my_did: DID,
+    relay_did: DID,
+    callback: Callable[[Message], Awaitable[None]] = None,
+) -> List[Message]:
+    """Fetch stored messages from the relay."""
+
+    # Fetch a count of all stored messages
+    message = Message(
+        type="https://didcomm.org/messagepickup/3.0/status-request",
+        id=str(uuid.uuid4()),
+        body={},
+        frm=my_did,
+        to=[relay_did],
+    )
+    message = await send_http_message(dmp, my_did, message, target=relay_did)
+
+    # If there's no messages, we can return early
+    if message.body["message_count"] == 0:
+        return []
+
+    # Request messages that are stored at the relay, according to the
+    # message_count returned in the previous call
+    message = Message(
+        type="https://didcomm.org/messagepickup/3.0/delivery-request",
+        id=str(uuid.uuid4()),
+        body={
+            "limit": message.body["message_count"],
+        },
+        frm=my_did,
+        to=[relay_did],
+    )
+    message = await send_http_message(dmp, my_did, message, target=relay_did)
+
+    # Handle each stored message is order we receive it
+    for attach in message.attachments:
+        LOG.info("Received message %s", attach["id"][:-58])
+
+        # Decrypt/Unpack the encrypted message attachment
+        unpacked = await dmp.packaging.unpack(json.dumps(attach["data"]["json"]))
```

Comment:
Feels like there should be a better way to do this; I'm pretty sure that dumped object will just immediately get `json.loads`ed on the other end. Reaching into `packaging` on `dmp` also feels odd. I think we should add methods and capabilities to dmp/packaging to handle this more cleanly.

Reply:
Reaching into packaging was a copy-paste thing that made it's way over here